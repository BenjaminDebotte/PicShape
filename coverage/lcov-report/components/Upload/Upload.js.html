<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Upload/Upload.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/Upload</a> Upload.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/24</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux'
&nbsp;
import { convertFile } from '../../actions/upload';
&nbsp;
&nbsp;
import './Upload.css';
&nbsp;
import request from 'superagent';
import Button from 'react-button';
&nbsp;
&nbsp;
export class Upload extends Component {
    constructor(props) {
        super(props);
        this.state = {
            buttonText: 'Browse',
            onLoad: 'False'
        }
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    _h</span>andleImageChange(input) {
<span class="cstat-no" title="statement not covered" >          console.log("_handleImageChange");</span>
<span class="cstat-no" title="statement not covered" >          input.preventDefault();</span>
&nbsp;
          let reader = <span class="cstat-no" title="statement not covered" >new FileReader();</span>
          let file = <span class="cstat-no" title="statement not covered" >input.target.files[0];</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >          reader.onloadend = <span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >              console.log('On load end');</span>
<span class="cstat-no" title="statement not covered" >              this.props.dispatch({</span>
                  type: 'LOAD_IMAGE_SUCCESS',
                  convertedImgLink: reader.result,
                  baseImg: reader.result
            });
        };
<span class="cstat-no" title="statement not covered" >          reader.readAsDataURL(file)</span>
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    _c</span>hangeImage(input) {
&nbsp;
<span class="cstat-no" title="statement not covered" >        if (document.getElementById("convertedImg").src === this.props.convertedImgLink) {</span>
<span class="cstat-no" title="statement not covered" >            document.getElementById("convertedImg").src = this.props.baseImg;</span>
        }
        else {
<span class="cstat-no" title="statement not covered" >            document.getElementById("convertedImg").src = this.props.convertedImgLink;</span>
        }
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    on</span>Click(event) {
<span class="cstat-no" title="statement not covered" >        event.preventDefault();</span>
&nbsp;
        var files = <span class="cstat-no" title="statement not covered" >document.getElementById('file_to_upload').files;</span>
        var iter = <span class="cstat-no" title="statement not covered" >document.getElementById('iter').value;</span>
        var mode = <span class="cstat-no" title="statement not covered" >document.getElementById('mode').value;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >        if (files.hasOwnProperty(0) &amp;&amp; files[0] instanceof File){</span>
<span class="cstat-no" title="statement not covered" >            this.props.dispatch(convertFile(files[0], {iter: iter, mode: mode}, this.props.token));</span>
        }
&nbsp;
}
&nbsp;
  render() {
     const thumbnail = (this.props.convertedImgLink !== '' ? (
                 &lt;a className="thumbnail"&gt;
                    &lt;img className="ui fluid image img" onClick={this._changeImage.bind(this)} id="convertedImg" src={this.props.convertedImgLink} width="100%" alt="" onMouseOver=""/&gt;
                 &lt;/a&gt;
         ) : (
<span class="branch-1 cbranch-no" title="branch not covered" >             &lt;div&gt;</span>
                  &lt;div className="fill img"&gt;
                      &lt;h2 className="preview"&gt;
                      Choose a picture !
                      &lt;/h2&gt;
                  &lt;/div&gt;
              &lt;/div&gt;
            )
     );
&nbsp;
     const valid_button = (
       &lt;button className="ui button" onClick={this.onClick.bind(this)} &gt;Send&lt;/button&gt;
&nbsp;
&nbsp;
     );
&nbsp;
&nbsp;
    return (
        &lt;div className="ui container"&gt;
        &lt;div className="ui raised segment"&gt;
            &lt;div className="row"&gt;
                    {thumbnail}
                    &lt;label className="ui button"&gt; {this.state.buttonText}
                        &lt;input className="form-control" onChange={this._handleImageChange.bind(this)} type='file' encType='multipart/form-data' id='file_to_upload' name='photo' style={{display: "none"}}/&gt;
                    &lt;/label&gt;
            &lt;/div&gt;
            &lt;div className="row"&gt;
                &lt;div className="center-block"&gt;
                    &lt;form className="ui form"&gt;
                        &lt;div className="field"&gt;
                            Iteration :
                            &lt;input className="form-control" id="iter" type="number" min="1" max="500"/&gt;
                        &lt;/div&gt;
                        &lt;div className="field"&gt;
                            Mode :
                            &lt;select id="mode"&gt;
                              &lt;option value="0"&gt;Combo&lt;/option&gt;
                              &lt;option value="1"&gt;Triangle&lt;/option&gt;
                              &lt;option value="2"&gt;Rectangle&lt;/option&gt;
                              &lt;option value="3"&gt;Ellipse&lt;/option&gt;
                              &lt;option value="4"&gt;Circle&lt;/option&gt;
                              &lt;option value="5"&gt;Rotatedrect&lt;/option&gt;
                              &lt;option value="6"&gt;Beziers&lt;/option&gt;
                              &lt;option value="7"&gt;Rotatedellipse&lt;/option&gt;
                              &lt;option value="8"&gt;Polygon&lt;/option&gt;
                            &lt;/select&gt;
                        &lt;/div&gt;
&nbsp;
&nbsp;
                        {valid_button}
&nbsp;
&nbsp;
                        &lt;/form&gt;
                &lt;/div&gt;
            &lt;/div&gt;
            &lt;/div&gt;
            &lt;/div&gt;
    );
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    token: state.auth.token,
    messages: state.messages,
    convertedImgLink: state.upload.convertedImgLink,
    baseImg: state.upload.baseImg
  };
};
&nbsp;
export default connect(mapStateToProps)(Upload);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Thu Mar 09 2017 10:50:14 GMT+0100 (Romance Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
