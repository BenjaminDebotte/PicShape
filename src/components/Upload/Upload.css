.upload {
  text-align: center;
}

.colonne {
    margin-top: 15px;
}

.colonne_2 {
    margin-top: 15px;
    padding-top: 15px;
    padding-bottom: 15px;
    cursor: pointer;
    height: 100%;
    border-width:2px;
    border-style:dashed;;
    border-color:black;
}

.fill {
  margin: 15%;
}
