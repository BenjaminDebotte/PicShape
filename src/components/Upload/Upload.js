import React, { Component } from 'react';
import { connect } from 'react-redux'

import { convertFile } from '../../actions/upload';


import './Upload.css';

import request from 'superagent';
import Button from 'react-button';


class Upload extends Component {
    constructor(props) {
        super(props);
        this.state = {
            buttonText: 'Browse',
            onLoad: 'False'
        }
    }

    _handleImageChange(input) {
          console.log("_handleImageChange");
          input.preventDefault();

          let reader = new FileReader();
          let file = input.target.files[0];

          reader.onloadend = () => {
              console.log('On load end');
              this.props.dispatch({
                  type: 'LOAD_IMAGE_SUCCESS',
                  convertedImgLink: reader.result,
                  baseImg: reader.result
            });
        };
          reader.readAsDataURL(file)
    }

    _changeImage(input) {
      if (input.currentTarget.src === this.props.convertedImgLink) {
          input.currentTarget.src = this.props.baseImg;
      }
      else {
          input.currentTarget.src = this.props.convertedImgLink;
      }
    }

    onClick(event) {
        event.preventDefault();

        var files = document.getElementById('file_to_upload').files;
        var iter = document.getElementById('iter').value;
        var mode = document.getElementById('mode').value;

        if (files.hasOwnProperty(0) && files[0] instanceof File){
            this.props.dispatch(convertFile(files[0], {iter: iter, mode: mode}, this.props.token));
        }

}

  render() {
     const thumbnail = (this.props.convertedImgLink !== '' ? (
                 <a className="thumbnail">
                    <img className="ui fluid image img" onClick={this._changeImage.bind(this)} id="convertedImg" src={this.props.convertedImgLink} width="100%" alt="" onMouseOver=""/>
                 </a>
         ) : (
             <div>
                  <div className="fill img"></div>
                      <h1>
                      Choose a picture !
                      </h1>

                  <div className="fill"></div>
              </div>
            )
     );

    return (
        <div className="ui container">
        <div className="ui raised center aligned segment">
            <div className="row">
                    {thumbnail}
                    <label className="ui button"> {this.state.buttonText}
                        <input className="form-control" onChange={this._handleImageChange.bind(this)} type='file' encType='multipart/form-data' id='file_to_upload' name='photo' style={{display: "none"}}/>
                    </label>
            </div>
            <div className="row">
                <div className="center-block">
                    <form className="ui form">
                        <div className="field">
                            Iteration :
                            <input className="form-control" id="iter" type="number" min="1" max="500"/>
                        </div>
                        <div className="field">
                            Mode :
                            <select id="mode">
                              <option value="0">Combo</option>
                              <option value="1">Triangle</option>
                              <option value="2">Rectangle</option>
                              <option value="3">Ellipse</option>
                              <option value="4">Circle</option>
                              <option value="5">Rotatedrect</option>
                              <option value="6">Beziers</option>
                              <option value="7">Rotatedellipse</option>
                              <option value="8">Polygon</option>
                            </select>
                        </div>

                        <button className="ui button" onClick={this.onClick.bind(this)} >Send</button>


                        </form>
                </div>
            </div>
            </div>
            </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    token: state.auth.token,
    messages: state.messages,
    convertedImgLink: state.upload.convertedImgLink,
    baseImg: state.upload.baseImg
  };
};

export default connect(mapStateToProps)(Upload);
