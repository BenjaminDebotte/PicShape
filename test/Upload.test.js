import React from 'react';
import  {Upload} from '../src/components/Upload/Upload';
import renderer from 'react-test-renderer';

describe('components', function() {
  describe('<Upload />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Upload />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });
});
