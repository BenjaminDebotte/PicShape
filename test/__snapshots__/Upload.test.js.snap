exports[`components <Upload /> renders correctly 1`] = `
<div
  className="ui container">
  <div
    className="ui raised segment">
    <div
      className="row">
      <a
        className="thumbnail">
        <img
          alt=""
          className="ui fluid image img"
          id="convertedImg"
          onClick={[Function bound _changeImage]}
          onMouseOver=""
          src={undefined}
          width="100%" />
      </a>
      <label
        className="ui button">
         
        Browse
        <input
          className="form-control"
          encType="multipart/form-data"
          id="file_to_upload"
          name="photo"
          onChange={[Function bound _handleImageChange]}
          style={
            Object {
              "display": "none"
            }
          }
          type="file" />
      </label>
    </div>
    <div
      className="row">
      <div
        className="center-block">
        <form
          className="ui form">
          <div
            className="field">
            Iteration :
            <input
              className="form-control"
              id="iter"
              max="500"
              min="1"
              type="number" />
          </div>
          <div
            className="field">
            Mode :
            <select
              id="mode">
              <option
                value="0">
                Combo
              </option>
              <option
                value="1">
                Triangle
              </option>
              <option
                value="2">
                Rectangle
              </option>
              <option
                value="3">
                Ellipse
              </option>
              <option
                value="4">
                Circle
              </option>
              <option
                value="5">
                Rotatedrect
              </option>
              <option
                value="6">
                Beziers
              </option>
              <option
                value="7">
                Rotatedellipse
              </option>
              <option
                value="8">
                Polygon
              </option>
            </select>
          </div>
          <button
            className="ui button"
            onClick={[Function bound onClick]}>
            Send
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
